fsc/hdf5_io/_save_load.py

```diff
@@ -50,9 +101,34 @@ def to_hdf5(obj, hdf5_handle):
     if hasattr(obj, 'to_hdf5'):
         obj.to_hdf5(hdf5_handle)
     else:
-        to_hdf5_singledispatch(obj, hdf5_handle)
-
-
+        try:
+            to_hdf5_singledispatch(obj, hdf5_handle)
+        except SerializerNotFound:
+            objtype = type(obj)
+            objmodule = objtype.__module__
+            if objmodule is None:
+                fullname = objtype.__qualname__
+            else:
+                fullname = objmodule + '.' + objtype.__qualname__
+
```

Comment:
could also be fixed here, before passing to _try_loading_parts

Reply:
For the fullname, it's a bit unfortunate that there is no way to "canonicalize" this, i.e. you can't figure out what the "public import name" of a class is, only its implementation location.

Reply:
Fair, that's a python feature I'm missing. Lets go with the full name in the mean time